docs/user/openstack/ovs-dpdk.md

```diff
@@ -0,0 +1,248 @@
+# Installing a cluster on OpenStack that supports DPDK-connected compute machines
+
+If the underlying OpenStack deployment has Open vSwitch with Data Plane Development Kit (OVS-DPDK) support enabled,
+an OpenShift cluster can take advantage of the feature by providing access to [poll mode drivers][1].
+
+## Pre-requisites
+
+The following steps might be required to be checked before starting the deployment of OpenShift.
+
+### Selecting hugepages size and CPU policy for OpenShift virtual machines
+
+You require hugepages for every OCP cluster virtual machine running on the hosts with OVS-DPDK.
+If hugepages are not present in the flavor, the machines will get created but the network interface won't work.
+You must check with your OpenStack administrator the size and number of available hugepages which can be used per node.
+
+This may be accomplished in two ways:
+
+- The flavors used for the OCP are defined with appropriate values. For example:
+
+    ``` bash
+    openstack flavor create m1.xlarge.nfv --ram 16384 --disk 40 --vcpus 8
+    openstack flavor set --property hw:cpu_policy=dedicated --property hw:mem_page_size=large m1.xlarge.nfv
+    ```
+
+    This method requires administrative access the OpenStack deployment because flavors need to be created and modified.
+
+- The image properties maybe be defined in the install-config.yaml using [platform.openstack.clusterOSImageProperties](customization.md).
+  For example:
+
+    ``` yaml
+    platform:
+      openstack:
+         clusterOSImageProperties:
+            - hw_cpu_policy: dedicated
+            - hw_mem_page_size: 1GB
+    ```
+
+If using image properties, then the flavor used to deploy the virtual machines must allow the image property values selected.
+For example, if you attempt to deploy a virtual machine using an image with the property `hw_mem_page_size: 1GB`,
+but the flavor being used is set with `--property hw:mem_page_size=small`,
+then the deployment of the virtual machine will fail because the flavor does not allow for using a 1GB page size (large).
+The flavor property would need to either not be set, which is the same as `--property hw:mem_page_size=any`,
+or the flavor property should be set to `--property hw:mem_page_size=large`.
+
+### Create an aggregate group (optional)
+
+If not all your OpenStack compute nodes are configured for OVS-DPDK, you must create an aggregate group and add the compatible nodes into it.
+
+Create the group, add the nodes, set the property and update the flavor:
+
+``` bash
+openstack aggregate create dpdk_group
+openstack aggregate add host dpdk_group <compute-hosts>
+openstack aggregate set --property dpdk=true dpdk_group
+openstack flavor set <flavor> --property dpdk=true
```

Comment:
This relies on (very) legacy behavior. You should prefix these properties with `aggregate_instance_extra_specs` to avoid conflicts with the `ComputeFilter`. From [the docs](https://docs.openstack.org/nova/latest/admin/aggregates.html#example-specify-compute-hosts-with-ssds)

> Once the flavor is created, specify one or more key-value pairs that match the key-value pairs on the host aggregates with scope `aggregate_instance_extra_specs`.

Reply:
thanks